src/core/public/application/ui/app_router.tsx

```diff
@@ -44,6 +45,8 @@ export const AppRouter: FunctionComponent<Props> = ({
   appStatuses$,
 }) => {
   const appStatuses = useObservable(appStatuses$, new Map());
+  const createScopedHistory = (basePath: string) => new ScopedHistory(history, basePath);
```

Comment:
Should we wrap it `useCallback`?

Reply:
I don't believe this is necessary since the history instance should be created each time an app is mounted. The app is only mounted when switching between apps, at which point we want to create a new scoped history instance.